### ISSUE TITLE
"-P exit" differs in behaviour than "-FS exit"

### ISSUE THREAD

**Issue by R00dNet** [ethereum-mining/ethminer#1405]
Please stop with all the changes. Ethminer was absolutely fine, ran and performed great.

-FS exit is sorely missed here at my end.. -P exit did not do this, as it was advised to do!

However with latest ethminer 16 dev2. A drop of speed of just over 1mhs for me. not impressed. Print out of speed and gpu speed (the same thing) what is purpose of reporting 2 speeds? why you feel the need to make changes that are not needed I dont get. etminer is, kinda was, a great miner. Now you messing around with it too much.

How you have managed to mess something up that was in such great condition I do not understand.

Just stop, get back to basics, sort it and leave it. This transition should have been simple.

**Comment by R00dNet** [comment]
no -P exit is not shifting batch file onto next algo nor is it behaving like -FS exit. Why do you think they have added this > failover-timeout this command now exits ethminer and onto next miner. without that ethminer just loops trying to reconnect with -P exit or without....

my batch is for multi algo on mph.

**Comment by StefanOberhumer** [comment]
1.) **Report an issue and use an appropriate subject and fill it with constructive content**

2.) 0.16.0.dev2 (remind the **dev2**) is a development snapshot - means current source compiled and bundled

> A drop of speed of just over 1mhs for me.

3.) If you would supply us with more info (OS, GPUS, ethminer --list-devices, ...) possible we can help better  getting back your missing 1MHs 

> Why do you think they have added this > failover-timeout this command now exits ethminer and onto next miner

4.) failover-timeout is the time after that ethminer tries to reconnect to the main pool which is the first -P parameter (you must not specify the failover-timeout parameter !)

5.) Maybe you can explain more detailed what has changed and make problems as I still not understand
Would be helpful to get (at least a part) of your batchfile to understand what's the problem

>Please stop with all the changes. Ethminer was absolutely fine, ran and performed great.

5.) If you like the old versions you can 
* download them at https://github.com/ethereum-mining/ethminer/releases
* clone the repo, go back in history and compile your preferred version

> Just stop, get back to basics, sort it and leave it. This transition should have been simple.

6.) Create PRs or Issues and provide us with sorted basic ideas

**Comment by R00dNet** [comment]
the changes required to compensate for the mhs loss makes no difference. as the speed would be and is faster on v15, with any increments in power to gpu than v16. i got speed back. at same settings is over 16mhs on v15.... gpu is rx480 8gb ref.

Al I want ethminer to do is what it did before and that is the -FS exit function. It is required to exit ethminer. I run a batch file that switches algos based on profitability, as i am sure a lot of other users do too. -P exit is not doing this for ethminer.

Im terribly sorry I think I made a mistake regarding failover-timeout. I am currently testing it out and waiting on an algo switch. Im sure it did it, not 100% though. my apologies.

**Comment by smurfy** [comment]
To have the miner exit (with ``-P exit``) imidently after a disconnect (of any kind) you need to use ``--farm-retries 0``  (i think thats the param) otherwise it tries to reconnect up to 3 times to reconnect to current pool before switching to new pool (which is in our case the exit command)

**Comment by R00dNet** [comment]
nope. ethminer just wont exit and move onto mext algo/miner in batch.
-P exit, --failover-timeout 0, --work-timeout 1, --exit, --response-timeout 2, --farm-retries 1

none of these work. just keeps looping trying to reconnect to ethash algo. and none do what -FS exit did.

![ethminer -p](https://user-images.githubusercontent.com/37847231/43646732-9b9b7fc2-972d-11e8-8101-19b83a2e5078.png)

**Comment by AndreaLanfranchi** [comment]
Can you repeat the very same test with 0.16 please ?

**Comment by AndreaLanfranchi** [comment]
And please remove --failover-timeout as it does not apply to your environment.

**Comment by AndreaLanfranchi** [comment]
Also would be helpful if you could post the "full" command line arguments.

>-P exit, --failover-timeout 0, --work-timeout 1, --exit, --response-timeout 2, --farm-retries 1

it's not clear if you tested those one by one or in some sort of combination.

**Comment by R00dNet** [comment]
that screenshot is from latest v16dev2. those commands were entered individually. they are a list not an order

**Comment by AndreaLanfranchi** [comment]
You should then test with

```
ethminer.exe -P stratum2+tcp://<whatever>@europe.ethash-hub.miningpoolhub.com:12020 -P exit --farm-retries 1

```

**Comment by R00dNet** [comment]
does that work for you? as it is not for me. ive exhausted all combos with the above. i will wait it out and see what v16 final is like. back to claymore for now :(

**Comment by AndreaLanfranchi** [comment]
Could you please report what is the output of 

ethminer.exe -V

I am asking as the output of your screen-shot does not appear to belong to 0.16

**Comment by R00dNet** [comment]
sure my mistake. same thing though.
![v16](https://user-images.githubusercontent.com/37847231/43651010-27999dfa-9739-11e8-87b7-04ee03f3b47e.png)

**Comment by AndreaLanfranchi** [comment]
Think I got where the problem is.
Please one more last test

```
ethminer.exe -P stratum2+tcp://<whatever>@europe.ethash-hub.miningpoolhub.com:12020 -P exit --farm-retries 0
```

**Comment by AndreaLanfranchi** [comment]
Nevermind ... it won't allow --farm-retries 0

**Comment by R00dNet** [comment]
did you test that before you posted it? I have done that. before i even came here to make this post. if you had tested it you would not have posted it. by looks of it you have not even read the help files.

im done with this for now.

**Comment by AndreaLanfranchi** [comment]
First of all please calm down.
We're here to help. I'm not here to undergo your rude attacks.

I'm one of the devs and just now I am addressing this specific issue. (In particular I'm the one who has made most of the changes in the stratum protocol detection).

So please, again, calm down and be patient.

**Comment by ddobreff** [comment]
./ethminer -G --report-hashrate --HWMON 0 -P stratum+tcp://mhubaccount.worker:x@europe.ethash-hub.miningpoolhub.com:20535 --api-bind -3333 --cl-local-work 64 -P exit
Results to this:
https://pastebin.com/QGbJTdaQ
I think u are using wrong port number.

**Comment by smurfy** [comment]
as far as i understand, what he wants is, that the miner exits after it gets disconnected from the pool for whatever reason. (pool closed, network error, whatever)

I think that was the behavior with ``-FS exit``.  So we should allow ``--farm-retries 0``  and hint about that within the ``-P exit``  command desc. I also would suggest renaming ``--farm-retries`` to something different (or alias it) like ``--pool-retries`` or something.

**Comment by ddobreff** [comment]
Correct, but this doesn't excuse the issuer's behaviour.

**Comment by R00dNet** [comment]
@ddofreff my behaviour? I am just tired with users like you who post nonsense. wrong port? really? what a waste of your time posting that message only to conclude that I am using wrong port.... you are wrong. 

@AndreaLanfranchi I am calm (do you see any caps or exclamation points?. was more bewildered than anything when I read your reply. you said to try something that does not work and is not a valid setting for that command....

@smurfy most sense relating to -FS exit function I have read. thanks

**Comment by smurfy** [comment]
> my behaviour?

Well your choice of words for the initial issue. Instead of filling out an issue ticket based on the template and describing the problem: ``-P exit`` does not work for your use-case, while ``-FS exit`` worked before, 
You did a bit of a rant :)

**Comment by AndreaLanfranchi** [comment]
@R00dNet ... there are many ways to appear rude and those do not necessarily imply usage of capital letters or exclamation points.

Everyone here uses his/her time on voluntary basis, not being paid, only guided by the desire to help and to improve. Full stop. We do not owe you anything and you don't pay anything for using this software.

This said I apologize for giving you a wrong advice but have you counted how many lines of code there are and **how many devs** contributing ? Sometimes, as we're all human, we make mistakes or, simply, do not recall well which was the correct syntax and/or range of values.

If you do not understand this please stay with Claymore which, as you pay a fee for it's usage, **should** give you a SLA. But I am pretty sure there's not !

This ticket is closed and on hold.

Bye.
